lib/ansible/modules/cloud/amazon/ec2_eip.py

```diff
@@ -159,300 +208,354 @@
   returned: on success
   type: str
   sample: eipalloc-51aa3a6c
+name:
+  description: name of the elastic ip
+  returned: on success
+  type: str
+  sample: myElasticIp-29
 public_ip:
   description: an elastic ip address
   returned: on success
   type: str
   sample: 52.88.159.209
+tags:
+  description: tags associated with the elastic ip
+  returned: on success
+  type: dict
+  sample:
+    key: value
+    key2: value2
 '''
 
 try:
-    import boto.exception
+    import botocore.exceptions
 except ImportError:
-    pass  # Taken care of by ec2.HAS_BOTO
+    pass  # Taken care of by ec2.HAS_BOTO3
+
+from ansible.module_utils.ec2 import (HAS_BOTO3, ec2_argument_spec, manage_tags, boto3_tag_list_to_ansible_dict)
 
-from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import HAS_BOTO, ec2_argument_spec, ec2_connect
+from ansible.module_utils.aws.core import AnsibleAWSModule
 
 
 class EIPException(Exception):
     pass
 
 
-def associate_ip_and_device(ec2, address, private_ip_address, device_id, allow_reassociation, check_mode, isinstance=True):
-    if address_is_associated_with_device(ec2, address, device_id, isinstance):
-        return {'changed': False}
+def is_instance(device_id):
+    if device_id.startswith("i-"):
```

Comment:
This is true for new-style long identifiers; weren't the old style identifiers like "ami-abc123"?

Reply:
Not for instances - I can only find references to "i-xxxx".  "ami-abc123" if for images.

Reply:
It may not be forward compatible then, but it seems unlikely AWS would change it unless they introduced something new.